src/config.ts

```diff
@@ -16,14 +24,29 @@ function parseList(list: string | undefined): string[] | undefined {
 }
 
 export function readConfig(): ConfigurationOptions {
-  const fail_on_severity = z
-    .enum(SEVERITIES)
-    .default('low')
-    .parse(getOptionalInput('fail-on-severity'))
+  const externalConfig = getOptionalInput('config-file')
+  if (externalConfig !== undefined) {
+    const config = readConfigFile(externalConfig)
+    // the reasoning behind reading the inline config when an external
+    // config file is provided is that we still want to allow users to
+    // pass inline options in the presence of an external config file.
+    const inlineConfig = readInlineConfig()
+    // the external config takes precedence
+    return Object.assign({}, inlineConfig, config)
```

Comment:
TIL about `Object.assign`, nicee.

Reply:
TBH I brought this from plain JS and I'm sure there's a prettier way to do this in 2022 with TS (@brphelps  @hmaurer maybe?)